nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/FetchHBaseRow.java

```diff
@@ -0,0 +1,407 @@
+            .name("Row Identifier")
+            .description("The identifier of the row to fetch.")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor COLUMNS = new PropertyDescriptor.Builder()
+            .name("Columns")
+            .description("An optional comma-separated list of \"<colFamily>:<colQualifier>\" pairs to fetch. To return all columns " +
+                    "for a given family, leave off the qualifier such as \"<colFamily1>,<colFamily2>\".")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.createRegexMatchingValidator(COLUMNS_PATTERN))
+            .build();
+
+    static final AllowableValue DESTINATION_ATTRIBUTES = new AllowableValue("flowfile-attributes", "flowfile-attributes",
+            "Adds each cell as a FlowFile attribute where they key is col-family:col-qualifier and the value is the cell's value.");
+    static final AllowableValue DESTINATION_CONTENT = new AllowableValue("flowfile-content", "flowfile-content",
+            "Overwrites the FlowFile content with a JSON document representing the row that was fetched. " +
+                    "The format of the JSON document is determined by the JSON Format property.");
+
+    static final PropertyDescriptor DESTINATION = new PropertyDescriptor.Builder()
+            .name("Destination")
+            .description("Indicates whether the row fetched from HBase is written to FlowFile content or FlowFile Attributes.")
+            .required(true)
+            .allowableValues(DESTINATION_ATTRIBUTES, DESTINATION_CONTENT)
+            .defaultValue(DESTINATION_ATTRIBUTES.getValue())
+            .build();
+
+    static final AllowableValue VALUE_ENCODING_STRING = new AllowableValue("string", "string", "Creates a String using the bytes of the cell value and the given Character Set.");
+    static final AllowableValue VALUE_ENCODING_BASE64 = new AllowableValue("base64", "base64", "Creates a Base64 encoded String of the cell value.");
+
+    static final PropertyDescriptor VALUE_ENCODING = new PropertyDescriptor.Builder()
+            .name("Value Encoding")
+            .description("Specifies how to represent the values of cells when stored in FlowFile attributes, or written to JSON.")
+            .required(true)
+            .allowableValues(VALUE_ENCODING_STRING, VALUE_ENCODING_BASE64)
+            .defaultValue(VALUE_ENCODING_STRING.getValue())
+            .build();
+
+    static final AllowableValue JSON_FORMAT_FULL_ROW = new AllowableValue("full-row", "full-row", "Creates a JSON document with the format: " +
+            "{\"row\": \"<row key>\", \"cells\": { \"<cell 1 family>:<cell 1 qualifier>\": \"<cell 1 value>\", \"<cell 2 family>:<cell 2 qualifier>\": \"<cell 2 value>\", ... }}.");
```

Comment:
Good point that we aren't doing anything with the timestamp. Based on that and the other comment about base64 encoding below, I'm now re-thinking the JSON format.

Currently the full-row option produces something like:
```{
  "row" : "row1",
  "cells": {
      "fam1:qual1" : "val1",
      "fam1:qual2" : "val2"
  }
}
```
And the qualifier-and-value option would produce:
```
{
    "qual1" : "val1",
    "qual2" : "val2"
}
```

The reason I added the qualifier-and-value option is because I expect someone will want to take the row and insert it into another system that already supports JSON, like MongoDB or Solr, and it might be a lot easier to have a flat JSON where its just field:value pairs.

I'm now thinking maybe the full row should be more like this:
```
{
  "row" : "row1",
  "cells": [
      {
        "family" : "fam1",
        "qualifier" : "qual1"
        "value" : "val1"
        "timestamp" : 123456789
      },
      {
        "family" : "fam1",
        "qualifier" : "qual2"
        "value" : "val2"
        "timestamp" : 123456789
      }
  ]
}
```
This way we include the timestamp, and also when we base64 encode it will seem more clear which piece is being looked at. 

We could pull the family up so that it isn't repeated, but I'm also trying to think about what people are going to do with this data after they get it from HBase, and I'm expecting someone may want to use SplitJSON to get each cell, and then they would potentially lose the family it came from if it wasn't repeated each time.

We could even go as far as to flatten the whole thing like:
```
[
  {
    "row" : "row1",
    "family" : "fam1",
    "qualifier" : "qual1"
    "value" : "val1"
    "timestamp" : 123456789
  },
  {
    "row" : "row1",
    "family" : "fam1",
    "qualifier" : "qual2"
    "value" : "val2"
    "timestamp" : 123456789
  }
]
```